src/pages/user/MyWebinars/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { useContext, useEffect, useMemo } from "react";
+import { useTranslation } from "react-i18next";
+import styled from "styled-components";
+import { useHistory } from "react-router-dom";
+import ProfileLayout from "@/components/Profile/ProfileLayout";
+import { EscolaLMSContext } from "@escolalms/sdk/lib/react/context";
+import { Tabs } from "@escolalms/components/lib/components/atoms/Tabs/Tabs";
+import ProfileWebinars from "@/components/Profile/ProfileWebinars";
+
+export enum WebinarStatus {
+  UPCOMING = "in_coming",
+  PAST = "is_ended",
+  STARTED = "is_started",
+}
+
+const MyWebinarsStyled = styled.section`
+  margin-top: -70px;
+  .tabs-menu {
+    margin: 0 0 70px 40px;
+  }
+`;
+
+const MyWebinarsPage = () => {
+  const { userWebinars, fetchUserWebinars } = useContext(EscolaLMSContext);
+  const { user } = useContext(EscolaLMSContext);
+  const history = useHistory();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    fetchUserWebinars();
+  }, [fetchUserWebinars]);
+
+  useEffect(() => {
+    if (!user.loading && !user.value) {
+      history.push("/login");
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const upcomingWebinars = useMemo(
+    () =>
+      userWebinars.list?.filter(
+        (webinar) => webinar.in_coming || webinar.is_started
+      ) || [],
+    [userWebinars.list]
+  );
+
+  const pastWebinars = useMemo(
+    () => userWebinars.list?.filter((webinar) => webinar.is_ended) || [],
+    [userWebinars.list]
+  );
```

Comment:
Its not big optimization but you could write some kind of sorting algorithm that would do it in one iteration. (e.g.)

```ts
const { upcomingWebinars, pastWebinars } = useMemo(
  () => 
    (userWebinars.list ?? []).reduce<{ upcomingWebinars: API.Webinar[]; pastWebinars: API.Webinar[] }>((acc, webinar) => {
    if (webinar.in_coming || webinar.is_started) acc.upcomingWebinars.push(webinar)
    if (webinar.is_ended) acc.pastWebinars.push(webinar)
    
    return acc; 
    }, { upcomingWebinars: [], pastWebinars: [] }))
```

I would probably also move whole algorithm to separate function to make the code self explanatory.